bin/impl/fetch.sh

```diff
@@ -17,36 +17,36 @@
 function download_verify() {
   url_prefix=$1
   tarball=$2
-  expected_md5=$3
+  expected_hash=$3
 
-  exp_md5_len=${#expected_md5}
-  if [[ "$exp_md5_len" -ne 32 ]]; then
-    echo "The expected MD5 checksum ($expected_md5) of $tarball has a length of $exp_md5_len but should be 32"
+  exp_hash_len=${#expected_hash}
+  if [[ "$exp_hash_len" -ne $HASH_LEN ]]; then
```

Comment:
could have a stricter sanity check here using a regex, verify its hex and of expected len.....